arm/resources/subscriptions/models.go

```diff
@@ -24,6 +24,35 @@ import (
 	"net/http"
 )
 
+// SpendingLimit enumerates the values for spending limit.
+type SpendingLimit string
+
+const (
+	// CurrentPeriodOff specifies the current period off state for spending
+	// limit.
+	CurrentPeriodOff SpendingLimit = "CurrentPeriodOff"
+	// Off specifies the off state for spending limit.
+	Off SpendingLimit = "Off"
+	// On specifies the on state for spending limit.
+	On SpendingLimit = "On"
+)
+
+// tate enumerates the values for tate.
+type tate string
```

Comment:
Found it. Its a new swagger and enum type name is `SubscriptionState` and package name is `Subsciptions`, therefore when we trim package name, it becomes `tate`. It was string type before so we did not do any stuttering type check earlier.